lib/xml-name-validator.js

```diff
@@ -1,17 +1,9 @@
 "use strict";
-const parser = require("./generated-parser.js");
 
 exports.name = potentialName => {
-  return mapResult(parser.startWith("Name").exec(potentialName));
+  return /^[:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF][:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\uD800-\uDB7F\uDC00-\uDFFF\-.0-9\u00B7\u0300-\u036F\u203F-\u2040]*$/u.test(potentialName);
```

Comment:
Do we need /u? Doesn't look like we make use of any of the fancy features.

Reply:
I've made it a policy to always use `/u` since it improves a number of semantics. Also I should be; it looks like https://www.w3.org/TR/REC-xml/#NT-Name goes beyond the 4 digit range. Will fix.